pkg/kubelet/prober/worker.go

```diff
@@ -222,6 +222,22 @@ func (w *worker) doProbe() (keepGoing bool) {
 			w.pod.Spec.RestartPolicy != v1.RestartPolicyNever
 	}
 
+	// Graceful shutdown of the pod.
+	if w.pod.ObjectMeta.DeletionTimestamp != nil {
+		// Set a last result to ensure quiet shutdown.
+		switch w.probeType {
+		case liveness, startup:
+			// Don't kill the container, it is already terminating.
+			w.resultsManager.Set(w.containerID, results.Success, w.pod)
+		case readiness:
+			// Stop sending traffic.
+			w.resultsManager.Set(w.containerID, results.Failure, w.pod)
+		}
+		// Stop probing at this point.
+		w.onHold = true
+		return false
```

Comment:
I don't know if we actually need to set `onHold`. It's only used here: https://github.com/kubernetes/kubernetes/blob/64a6ee4321d37740b84a5c5c7aaa67e0bf6b6508/pkg/kubelet/prober/worker.go#L209

But we're returning `false` so the worker for this probe will never run again anyways.

Reply:
True, I forgot the container won't probably restart if we asked for the pod to be deleted... removing it.